### PR TITLE
[AIRFLOW-4844] Added optional is_paused_upon_creation argument for da…

### PR DESCRIPTION
**Pull request by Sinsin1367**
…gs to configure initial pause status upon creation.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow-4844](https://issues.apache.org/jira/browse/AIRFLOW-4844)

### Description

- [ ] Here are some details about my PR:
1. Adding the feature to configure each dag object with the initial state of is_paused flag
2. The corresponding changes for python2 branch will be submitted once this review is finalized.

### Tests

- [ ] My PR adds the following unit tests:
1. Adding unit test to test existing dags in database not being affected with the flag.
2. Adding unit test to test new dags following the flag passed. (there are already unit tests testing backward compatibility of the the code change.)

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/models/dag.py [hunk 1]
```diff
@@ -160,6 +160,10 @@ class DAG(BaseDag, LoggingMixin):
     :param access_control: Specify optional DAG-level permissions, e.g.,
         "{'role1': {'can_dag_read'}, 'role2': {'can_dag_read', 'can_dag_edit'}}"
     :type access_control: dict
+    :param is_paused_upon_creation: Specifies if the dag is paused when created for the first time.
+        If the dag exists already, this flag will be ignored. If this optional parameter
+        is not specified, default behaviour same as before is exercised.
+    :type is_paused_upon_creation: bool
```
Comment (ashb) [inline airflow/models/dag.py]:
```suggestion
    :type is_paused_upon_creation: bool or None
```

File: airflow/models/dag.py [hunk 2]
```diff
@@ -186,7 +190,8 @@ def __init__(
         on_failure_callback: Optional[Callable] = None,
         doc_md: Optional[str] = None,
         params: Optional[Dict] = None,
-        access_control: Optional[Dict] = None
+        access_control: Optional[Dict] = None,
+        is_paused_upon_creation: Optional[bool] = None
```
Comment (ashb) [inline airflow/models/dag.py]:
```suggestion
        is_paused_upon_creation: Optional[bool] = None,
```

**Review by ashb — CHANGES_REQUESTED** [review]
Sounds useful.

File: airflow/models/dag.py [hunk 3]
```diff
@@ -160,6 +160,10 @@ class DAG(BaseDag, LoggingMixin):
     :param access_control: Specify optional DAG-level permissions, e.g.,
         "{'role1': {'can_dag_read'}, 'role2': {'can_dag_read', 'can_dag_edit'}}"
     :type access_control: dict
+    :param is_paused_upon_creation: Specifies if the dag is paused when created for the first time.
+        If the dag exists already, this flag will be ignored. If this optional parameter
+        is not specified, default behaviour same as before is exercised.
```
Comment (ashb) [inline airflow/models/dag.py]:
```suggestion
        is not specified, the global config setting will be used.
```

File: airflow/models/dag.py (hunk 1 above)
Comment (Sinsin1367) [reply airflow/models/dag.py]: Done.

File: airflow/models/dag.py (hunk 2 above)
Comment (Sinsin1367) [reply airflow/models/dag.py]: Done.

File: airflow/models/dag.py (hunk 3 above)
Comment (Sinsin1367) [reply airflow/models/dag.py]: Done.

**Review by ashb — APPROVED** [review]

**Pull request merged**